instrumentation/cassandra/cassandra-3.0/src/main/java/io/opentelemetry/instrumentation/auto/cassandra/v3_0/TracingSession.java

```diff
@@ -150,7 +144,7 @@ public ResultSetFuture executeAsync(String query) {
     Span span = TRACER.startSpan(session, query);
     try (Scope ignored = TRACER.startScope(span)) {
       ResultSetFuture future = session.executeAsync(query);
-      future.addListener(createListener(span, future), EXECUTOR_SERVICE);
+      future.addListener(createListener(span, future), SameThreadExecutor.INSTANCE);
```

Comment:
Probably not for this PR but should use `Futures.addCallback` to avoid recording a wrapped exception.